FWCore/ParameterSet/python/Modules.py

```diff
@@ -389,6 +389,10 @@ def insertInto(self, parameterSet, myname):
         newpset.addString(True, "@module_edm_type", "EDProducer")
         newpset.addVString(True, "@all_cases", [myname+"@"+p for p in self.parameterNames_()])
         newpset.addString(False, "@chosen_case", myname+"@"+self._chooseCase())
+        # The process name is tracked already in the process PSet, no
+        # need to keep tracked here. This way we can also transparently
+        # change the delivery of the process name to SwitchProducer.
+        newpset.addString(False, "@process_name", parameterSet.getString(True, "@process_name"))
```

Comment:
I'm not really happy about this solution, but I wasn't able to find an existing way to get the current process name (or process ParameterSet) from an EDModule constructor.

Reply:
It looks like `Factory::makeModule` is passed a `MakeModuleParams` which holds a `ProcessConfiguration` and inside that is the process name. You could inject the info at the time we do the `fillDescriptions`

https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/src/WorkerMaker.cc#L62

Reply:
Thanks, I'll take a look.

Reply:
@Dr15Jones Were you thinking I would add the special untracked string `@process_name` for all modules? (in principle limiting to `SwitchProducer` should be straightforward).

I was also thinking to ask how would I modify the PSet of the module there given that `MakeModuleParams` is taken as const reference, but I see that it a pointer to the `ParameterSet` and that is used to get a pointer to non-const via the const reference. Is this circumventing `const` actually intentional here?

Reply:
I thought you had intended to do it for all modules. I agree that only for SwitchProducer would be best. We create a new PSet anyway when we run the fillDescriptions.

Reply:
The python here adds it only for the `SwitchProducer`, I'll then replicate that behavior in the `Factory::makeModule()` (even if it means more code).